skins/oasis/modules/CommentsLikesController.class.php

```diff
@@ -75,40 +72,37 @@ private function getCommentsTooltip() {
 	 * @param Integer $count - number of comments/talk pages
 	 * @return String - formatted count
 	 */
-	private function formatCount($count) {
-		if ($count > 999999) {
-			return $formattedComments = wfMessage('oasis-page-header-comments-m', floor($count / 1000000))->text();
-		}
-		else if ($count > 999) {
-			return $formattedComments = wfMessage('oasis-page-header-comments-k', floor($count / 1000))->text();
+	private function formatCount( $count ) {
+		if ( $count > 999999 ) {
+			return $formattedComments = wfMessage( 'oasis-page-header-comments-m', floor( $count / 1000000 ) )->escaped();
+		} else if ( $count > 999 ) {
+			return $formattedComments = wfMessage( 'oasis-page-header-comments-k', floor( $count / 1000 ) )->escaped();
 		}
 		return $count;
 	}
 
-	public function executeIndex($data) {
-		wfProfileIn(__METHOD__);
-		global $wgTitle, $wgContentNamespaces, $wgExtraNamespacesLocal;
+	public function index( $data ) {
+		wfProfileIn( __METHOD__ );
 
-		if(empty($wgExtraNamespacesLocal)){
-			$wgExtraNamespacesLocal = array();
+		if ( empty( $this->wg->ExtraNamespacesLocal ) ) {
+			$this->wg->ExtraNamespacesLocal = [];
 		}
 
 		// set the page for which we're showing comments / likes
 		// used for proper linking on blog posts listings
-		if (!empty($data['title'])) {
-			$this->contextTitle = $data['title'];
-		}
-		else {
+		if ( $this->request->getVal( 'title' ) ) {
+			$this->contextTitle = $this->request->getVal( 'title' );
+		} else {
 			// by default we're showing # of comments for current page
-			$this->contextTitle = &$wgTitle;
+			$this->contextTitle = &$this->wg->Title;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)